doc/source/rllib/rllib-learner.rst

```diff
@@ -0,0 +1,459 @@
+            # call to `async_update`
+            results = learner_group.async_update(DUMMY_BATCH)
+
+            # This is an additional non-gradient based update.
+            learner_group.additional_update(**ADDITIONAL_UPDATE_KWARGS)
+
+        When updating a `LearnerGroup` you can perform blocking or async updates on batches of data. Async updates are necessary for implementing async algorithms such as APPO/IMPALA.
+        You can perform non-gradient based updates using `additional_update`.
+
+    .. tab-item:: Updating a Learner
+
+        .. testcode::
+            :skipif: True
+
+            # This is a blocking update.
+            result = learner.update(DUMMY_BATCH)
+
+            # This is an additional non-gradient based update.
+            learner_group.additional_update(**ADDITIONAL_UPDATE_KWARGS)
+
+        When updating a `Learner` you can only perform blocking updates on batches of data.
+        You can perform non-gradient based updates using `additional_update`.
+    
+
+Getting and setting state
+-------------------------
+
+.. tab-set::
+    
+    .. tab-item:: Getting and Setting State for a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            # module weights and optimizer states
+            state = learner_group.get_state()
+            learner_group.set_state(state)
+
+            # just module weights
+            weights = learner_group.get_weights()
+            learner_group.set_weights(weights)
+
+        Set/get the state dict of all learners through learner_group via `set_state` or 
+        `get_state`. This includes all states including both neural network weights, 
+        and optimizer states on each learner.
+    
+    .. tab-item:: Getting and Setting State for a Learner
+
+        .. testcode::
+            :skipif: True
+
+            # module weights and optimizer states
+            state = learner.get_state()
+            learner.set_state(state)
+
+            # just module weights
+            weights = learner.module.get_state()
+            learner.module.set_state(weights)
+
+        You can set and get the weights of a `Learner` using `set_state` and `get_state` .
+        For setting or getting only RLModule weights (without optimizer states), use `set_weight` or `get_weight` API.
+
+
+.. testcode::
+	:hide:
+    :skipif: True
+
+	import shutil
+	import tempfile
+
+	LEARNER_CKPT_DIR = str(tempfile.TemporaryDirectory())
+	LEARNER_GROUP_CKPT_DIR = str(tempfile.TemporaryDirectory())
+
+
+Checkpointing
+-------------
+
+.. tab-set::
+    
+    .. tab-item:: Checkpointing a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            learner_group.save_state(LEARNER_GROUP_CKPT_DIR)
+            learner_group.load_state(LEARNER_GROUP_CKPT_DIR)
+        
+        Checkpoint the state of all learners in the `LearnerGroup` via `save_state` and
+        `load_state`. This includes all states including neural network weights and any
```

Comment:
Can we make clear what the difference is between get/set/save/load state?
For many of these sections, we just start off with a code example without giving context.
I think both sections "Getting and setting state" and "Checkpointing" should begin with something like:

> "You can checkpoint the state of a LearnerGroup to a directory by calling `LearnerGroup.save_state()`. This will call `LearnerGroup.get_state()` under the hood and therefore includes neural networks weights and optimizer states. 
> Loading a checkpoint works accordingly. Here is an example of how to save and load the state of a LearnerGroup:

```
.. testcode::
            :skipif: True

            learner_group.save_state(LEARNER_GROUP_CKPT_DIR)
            learner_group.load_state(LEARNER_GROUP_CKPT_DIR)
```
> Note that since the state of all of the `Learner` s is identical, only the states from the first `Learner` need to be saved.

Reply:
done.